### PR TITLE
feat(autoalign): @floating-ui/dom implementation

### LINKED ISSUE #11506: [CWC]: Implement Floating-UI to Popover components
### Description

There's an issue with Popover being clipped when placed in a parent container with overflow set. This ticket tracks the implementation of the [`floating-ui` library ](https://floating-ui.com/docs/tutorial) in order to try to fix this issue.

Screenshot 2024-02-07 at 10 11 36 AM

### Component(s) impacted

popover, toogle-tip, slug

### Package

@carbon/web-components

### CodeSandbox example

https://stackblitz.com/edit/web-platform-w4k18m?file=index.html,styles.css

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Closes #11506

### Description

This PR adds the [`@floating-ui/dom` library](https://floating-ui.com) to address the clipping issue seen with adopters when placing the `popover`/`tooltip`/`toogletip`/`slug` in a parent container with `overflow:hidden` set.

Screenshot 2024-02-07 at 10 11 36 AM

The changes will be made under the `autoalign` property so the original components are not affected.

AFTER:
Screenshot 2024-02-17 at 3 13 24 PM

### Changelog

**New**

- `@floating-ui/dom` added to dependencies
- `floating-ui` global internal util
- Codesandbox example for `toggletip`
- `floating-ui` internal global utility to handle the positioning in `popover` and `toggletip`
- Codesandbox example for the components with the `autoalign` feature
- Storybook for autoalign in the experimental section
- Storybook docs for autoalign story

**Changed**

- `toggletip` story docs updated
- `popover`, `toggletip`, `tooltip`, `icon-button`, `slug` have the `autoalign` property added

### CODE REVIEW

File: packages/carbon-web-components/src/components/slug/slug.scss [hunk 1]
```diff
@@ -315,6 +315,121 @@ $colorMap: (
...
+
+/* stylelint-disable-next-line no-duplicate-selectors */
+:host(
+    #{$prefix}-slug[autoalign]:not([kind='hollow']):not([dot-type='hollow'])
+  ) {
+  &[alignment='top'],
+  &[alignment='top-left'],
+  &[alignment='top-right'],
+  &[alignment='top-start'],
+  &[alignment='top-end'] {
```
Comment (ariellalgilmore) [inline packages/carbon-web-components/src/components/slug/slug.scss:343]: could these styles be? 

```suggestion
  &[alignment^='top'], {
```

Comment (annawen1) [reply packages/carbon-web-components/src/components/slug/slug.scss:343]: Ah yes!

File: packages/carbon-web-components/src/components/toggle-tip/toggletip.scss [hunk 2]
```diff
@@ -10,6 +10,7 @@
 @use '@carbon/styles/scss/config' as *;
 @use '@carbon/styles/scss/type' as *;
 @use '@carbon/styles/scss/utilities';
+@use '@carbon/styles/scss/utilities/custom-property';
```
Comment (ariellalgilmore) [inline packages/carbon-web-components/src/components/toggle-tip/toggletip.scss:13]:
```suggestion
@use '@carbon/styles/scss/utilities' as *;
```

and then maybe we can remove the `@include custom-property.declaration(` to `@include declaration(`

**Review by ariellalgilmore — APPROVED** [review]
LGTM! just wonder if we should update all the examples the slug examples to use `autoalign` too or the readme's assuming peopl

**Comment by annawen1** [comment]
> LGTM! just wonder if we should update all the examples the slug examples to use `autoalign` too or the readme's assuming people copy and paste that. Also need to redirect to the release branch

@ariellalgilmore I did consider changing all the slug examples to use the `autoalign` but I think we should give adopters time to test it out since it is a new library before we push it forward

**Pull request merged**